packages/effect/src/Effect.ts

```diff
@@ -6277,3 +6336,240 @@ export const Tag: <const Id extends string>(id: Id) => <
     })
     return done
   }
+
+/**
+ * @since 2.0.0
+ * @category models
+ */
+export declare namespace Service {
+  /**
+   * @since 2.0.0
+   * @category models
+   */
+  export interface ProhibitedType {
+    Service?: `property "Service" is forbidden`
+    Identifier?: `property "Identifier" is forbidden`
+    Layer?: `property "Layer" is forbidden`
+    Default?: `property "Layer" is forbidden`
+    _op_layer?: `property "_op_layer" is forbidden`
+    _op?: `property "_op" is forbidden`
+    _tag?: `property "_tag" is forbidden`
+    of?: `property "of" is forbidden`
+    context?: `property "context" is forbidden`
+    key?: `property "key" is forbidden`
+    stack?: `property "stack" is forbidden`
+    name?: `property "name" is forbidden`
+    pipe?: `property "pipe" is forbidden`
+    use?: `property "use" is forbidden`
+  }
+
+  /**
+   * @since 2.0.0
+   * @category models
+   */
+  export type AllowedType<P> = Record<PropertyKey, any> & P
+
+  /**
+   * @since 2.0.0
+   * @category models
+   */
+  export type WithDependencies<O> =
+    | Simplify<O & { dependencies?: never }>
+    | Simplify<O & { dependencies: [] }>
+    | Simplify<O & { dependencies: [Layer.Layer.Any, ...Array<Layer.Layer.Any>] }>
+
+  /**
+   * @since 2.0.0
+   * @category models
+   */
+  export type Maker<P> =
+    | WithDependencies<{ effect: Effect<AllowedType<P>, any, any> }>
+    | WithDependencies<{ scoped: Effect<AllowedType<P>, any, any> }>
+    | WithDependencies<{ sync: () => AllowedType<P> }>
+
+  /**
+   * @since 2.0.0
+   * @category models
+   */
+  export type Options =
+    | (Service.Maker<Service.ProhibitedType> & { accessors: true })
+    | (Service.Maker<{}> & { accessors?: false })
+
+  /**
+   * @since 2.0.0
+   * @category models
+   */
+  export interface TagClass<Self, Id, Type> extends Context.Tag<Self, Self> {
+    make(_: Type): Self
```

Comment:
`make(this: void, _: Type): Self`

Reply:
why?

Reply:
So I can Effect.andThen(x, this.make) without an ESLint error about binding.